src/iaphandler.cpp

```diff
@@ -0,0 +1,271 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "iaphandler.h"
+#include "constants.h"
+#include "leakdetector.h"
+#include "logger.h"
+
+#include <QCoreApplication>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QScopeGuard>
+
+#ifdef MVPN_IOS
+#  include "platforms/ios/iosiaphandler.h"
+#endif
+
+namespace {
+Logger logger(LOG_IAP, "IAPHandler");
+IAPHandler* s_instance = nullptr;
```

Comment:
@bakulf I followed your idea below for the createInstance but I'm a little unsure about how / whether an s_instance will actually be available at runtime - and can't test :(